src/Microsoft.Graph.Core/Requests/Content/BatchRequestContent.cs

```diff
@@ -153,6 +152,25 @@ public bool RemoveBatchRequestStepWithId(string requestId)
             return isRemoved;
         }
 
+        /// <summary>
+        /// Creates a new <see cref="BatchRequestContent"/> with all <see cref="BatchRequestStep"/> that failed.
+        /// </summary>
+        /// <param name="responseStatusCodes">A dictionary with response codes, get with batchResponseContent.GetResponsesStatusCodesAsync()</param>
+        /// <returns>new <see cref="BatchRequestContent"/> with all failed requests.</returns>
+        public BatchRequestContent NewBatchWithFailedRequests(Dictionary<string, HttpStatusCode> responseStatusCodes)
```

Comment:
Any chance it makes sense to make `BatchResponseContent.IsSuccessStatusCode` an optional functional parameter that defaults to it so that it can be changed?
I suspect that automatically filtering every status code above 299 may not be enough and a user may want to pass in their own lambda to retry only specific codes they know if what they are looking out for.

Reply:
I totally agree with that we have to do something with that, but I was struggling with what was the best way. With the code in this PR, you can already do something like this before calling that method, if you don't want to retry all faulty status codes.

```csharp
responseStatuses.Where(r => r.Value == (HttpStatusCode)429);
```

My other idea was to use
```csharp
public BatchRequestContent NewBatchWithFailedRequests(Dictionary<string, HttpStatusCode> responseStatusCodes, params HttpStatusCode retryStatuses) {...}
```

But because the library targets `netstandard2.0` the `HttpStatusCode.TooManyRequests` is not available every where.

Open for suggestions, let's pick the best one.